### PR TITLE
10709 implementing query part of grouping

### PR DESCRIPTION
**Pull request by shreyan13680**
Code changes to implement grouping in filter expression
Unit Test cases for the code changes done

### CODE REVIEW

**Review by bhalcha13813 — APPROVED** [review]
Looks good

File: src/app/core/services/fss-search-filter.service.ts [hunk 1]
```diff
@@ -15,24 +15,28 @@ export class FssSearchFilterService {
...
 
       //Append join operator from second search condition
-      if (i != 0) {
-        filter = filter.concat(' ', fssSearchRow.selectedJoinOperator, ' ');
+      if(rowIndex != 0) {
+        filter = filter.concat(' ',fssSearchRow.selectedJoinOperator, ' ');        
       }
-      if (fieldDataType === this.stringDataType || fieldDataType === this.attributeDataType) {
-        if (operaterType === this.typeOperator) {
+      //Append opening barckets for grouping query.
```
Comment (martyn-fewtrell) [inline src/app/core/services/fss-search-filter.service.ts:34]: small typo barckets should be brackets?

File: src/app/core/services/fss-search-filter.service.ts [hunk 2]
```diff
@@ -50,13 +54,15 @@ export class FssSearchFilterService {
...
         if (operaterType === this.typeOperator) {
           const value = new Date(fssSearchRow.value + ' ' + fssSearchRow.time).toISOString();
-          console.log(new Date(fssSearchRow.value + ' ' + fssSearchRow.time), value);
           filter = filter.concat(fssSearchRow.selectedField, " ", fssSearchRow.selectedOperator, " ", value);
         }
         else if (operaterType === this.nullOperatorType) {
           filter = filter.concat(fssSearchRow.selectedField, " ", fssSearchRow.selectedOperator);
         }
       }
+      //Append closing barckets for grouping query
```
Comment (martyn-fewtrell) [inline src/app/core/services/fss-search-filter.service.ts:63]: Likewise brackets in comments

**Review by martyn-fewtrell — COMMENTED** [review]
A couple of very minor typos but otherwise good to go

File: src/app/core/services/fss-search-filter.service.ts (hunk 1 above)
Comment (shreyan13680) [reply src/app/core/services/fss-search-filter.service.ts]: corrected the typo

File: src/app/core/services/fss-search-filter.service.ts (hunk 2 above)
Comment (shreyan13680) [reply src/app/core/services/fss-search-filter.service.ts]: corrected the typo

**Review by martyn-fewtrell — APPROVED** [review]
Thanks for the update

**Pull request merged**